### PR TITLE
feat(components): use DS Error State in Stepper

### PR DESCRIPTION
**Pull request by romainseb**
**What is the problem this PR is trying to solve?**
Migrate from the old error feedback to the ErrorState from the DS

CleanShot 2023-03-17 at 17 32 29@2x

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/Stepper/Stepper.component.js [hunk 1]
```diff
@@ -170,6 +172,16 @@ function Stepper({ steps, title, renderActions, children }) {
 		}
 	}, [steps]);
 
+	if (isInError) {
```
Comment (jmfrancois) [inline packages/components/src/Stepper/Stepper.component.js:175]: The isInError is still used after, could you remove the usage on the code after this if branch ?

**Review by jmfrancois — CHANGES_REQUESTED** [review]

File: packages/components/src/Stepper/Stepper.component.js (hunk 1 above)
Comment (romainseb) [reply packages/components/src/Stepper/Stepper.component.js:175]: Yes 👍🏼

File: packages/components/src/Stepper/Stepper.component.test.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { render } from '@testing-library/react';
 import Stepper from './Stepper.component';
```
Comment (jmfrancois) [inline packages/components/src/Stepper/Stepper.component.test.js:4]: may be ?
```suggestion
jest.unmock('@talend/design-system')
```

File: packages/components/src/Stepper/__snapshots__/Stepper.component.test.js.snap [hunk 3]
```diff
@@ -1,179 +1,126 @@
...
-               (Aborted)
-            </div>
-          </li>
-        </ol>
-      </div>
-    </div>
-  </StepperTransition>
-</React.Fragment>
+      <span
+        class="CoralErrorState"
```
Comment (jmfrancois) [inline packages/components/src/Stepper/__snapshots__/Stepper.component.test.js.snap:13]: this is a mock of ds

File: packages/components/src/Stepper/Stepper.component.test.js (hunk 2 above)
Comment (romainseb) [reply packages/components/src/Stepper/Stepper.component.test.js:4]: I wasn't sure of if we want to unmock the DS in this package.
I add it so

File: packages/components/src/Stepper/Stepper.component.js [hunk 4]
```diff
@@ -170,25 +172,28 @@ function Stepper({ steps, title, renderActions, children }) {
...
+		const errorStep = getStepInError(steps);
```
Comment (aamalric-talend) [inline packages/components/src/Stepper/Stepper.component.js:176]: Can it be possible that we get 'undefined' here, let's say there's no status LOADING_STEP_STATUSES.FAILURE ??

**Review by aamalric-talend — APPROVED** [review]

**Review by jmfrancois — APPROVED** [review]

**Pull request merged**